skein/src/lib.rs

```diff
@@ -0,0 +1,259 @@
+//! Implementation of Skein cryptographic hash algorithms.
+//! The Skein hash function was one of the submissions to SHA-3,
+//! the cryptographic hash algorithm competition organized by the NIST.
+//!
+//! There are 3 standard versions of the Skein hash function:
+//!
+//! * `Skein-256`
+//! * `Skein-512`
+//! * `Skein-1024`
+//!
+//! # Examples
+//!
+//! Output size of Skein-256 is fixed, so its functionality is usually
+//! accessed via the `Digest` trait:
+//!
+//! ```
+//! use digest::generic_array::typenum::U32;
+//! use hex_literal::hex;
+//! use skein::{Digest, Skein256};
```

Comment:
It's better to use re-exported `digest` in user-facing examples:
```rust
use skein::{Digest, Skein256, digest::generic_array::typenum::U32};
```

You also could re-export [`digest::consts`](https://docs.rs/digest/latest/digest/consts/index.html) for additional convenience:
```rust
use skein::{Digest, Skein256, consts::U32};
```